### DISCUSSION TITLE
Static site generation

### DISCUSSION THREAD

**Discussion by bradbeattie** [redimp/otterwiki#116] [Ideas]
Curious if there'd be interest in the ability to export an otterwiki to static html. I'm mulling on updating a static site I host, and having used otterwiki for other things I'd like to use it here too. I've grown fairly comfortable with the write/browse workflow.

Were I to issue a pull request for this, would this be welcome?

**Comment by redimp** [comment]
Hey @bradbeattie, I think this would be a useful feature, a PR would be welcome.

Got a question: How would you handle the features which are difficult to implement in a meaningful way, like the diff for a page history?

**Reply by bbeattie-phxlabs** [reply]
My initial intent was to not include any of the page history, but instead only a snapshot of what currently exists. Likely the easiest to implement initially, and admittedly enough for my usecase.

**Reply by redimp** [reply]
I also think that a simple approach would make sense. The only important thing is that what comes out is usable .. and for most cases, the most recent version is what both is what people want to be accessible and what visitors want to read.

**Reply by rnbr04** [reply]
I was able to create a static site using `wget`. But it had visible options and pages which are not meant for public view in my case. Maybe the future plans involving fine-grained permissions for pages would help mitigate this problem.

Related issues: #54 #195
